week-02-ajax-and-js-objects/day-02/module-02/readme.md

```diff
@@ -0,0 +1,206 @@
+
+#### How to Interact with APIs
+
+1. `curl` in the terminal
+
+  ```zsh
+  ➜  curl -i "https://api.spotify.com/v1/artists/3jOstUTkEu2JkjvRdBA5Gu"
+  ```
+
+2. RESTful API Explorer - <a href="https://chrome.google.com/webstore/detail/postman-rest-client-short/mkhojklkhkdaghjjfdnphfphiaiohkef" target="_blank">Postman</a>
+
+  ```
+  https://api.spotify.com/v1/artists/3jOstUTkEu2JkjvRdBA5Gu
+  ```
+
+3. jQuery AJAX
+
+  ```js
+  $.get('https://api.spotify.com/v1/artists/3jOstUTkEu2JkjvRdBA5Gu', function(data) {
+    console.log(data);
+  });
+  ```
+
+#### Structure of a URL
+
+![url-structure](https://cloud.githubusercontent.com/assets/7833470/10722057/71282db8-7b69-11e5-867e-df5ef301ab19.png)
+
+| Spotify Query Type | Spotify Example URL |
+| :--- | :--- |
+| Artist Lookup by ID | `https://api.spotify.com/v1/artists/3jOstUTkEu2JkjvRdBA5Gu` |
+| Artist Search | `https://api.spotify.com/v1/search?q=weezer&type=artist` |
+
+#### Structure of a Response
+
+![response](https://cloud.githubusercontent.com/assets/7833470/10722166/95394272-7b6a-11e5-9679-3381c5baeaee.gif)
+
+## AJAX
+
+Asynchronous JavaScript And XML (AJAX) allows us to make requests to a server (ours or another application's) without refreshing the page.
+
+#### Why do we care?
+
+* AJAX lets us exchange data with the server behind the scenes. We can update our web pages (and data on the server!) without reloading the page at all.
+* Limiting page reloads makes our web apps *faster* and gives our users a *better experience*. (Imagine if you experienced a full page refresh every time you "liked" a post on Facebook!)
+
+#### How do we use it?
+
+jQuery gives us a <a href="https://api.jquery.com/category/ajax" target="_blank">list of methods</a> for making AJAX requests.
+
+## GET & POST
+
+The HyperText Transfer Protocol (HTTP) is similar to a written language like English. English was made for humans; the HTTP language is specifically for web browsers and servers to communicate with each other.
+
+`GET` and `POST` are the most important verbs in HTTP:
+
+* A browser will use `GET` to indicate it would like to receive a specific web page or resource from a server.
+* A browser will use `POST` to indicate it would like to send some data to a server.
+
+We can use AJAX to make both `GET` and `POST` requests to servers. jQuery gives us the <a href="https://api.jquery.com/jQuery.ajax" target="_blank">$.ajax()</a> method, which will allow us to perform any AJAX request. It also gives us the helper methods <a href="https://api.jquery.com/jQuery.get" target="_blank">$.get()</a> and <a href="https://api.jquery.com/jQuery.post" target="_blank">$.post()</a>, which, you guessed it, are specifically for `GET` and `POST` requests.
+
+## AJAX Setup
+
+Using jQuery's `$.ajax()` method, we can specify a list of parameters, including:
+
+* type of request
+* request URL
+* data type
+* callback function (which will run on successful completion of the AJAX request)
+
+```js
+$.ajax({
+  type: 'GET',
```

Comment:
Since we're using JSON notation without stringifying the key name, it may be worth casually mentioning that some names aren't allowed unless they are a string and that if they write JSON the format is invalid.

For instance:

```js
var dict = {my-Object: true}; // Raises syntax error
var dict = {"my-Object": true}; // OK
```

I was thinking about this because they will be introduced to the JSON response format and the JSON response format doesn't allow unquoted string key names :/

```js
JSON.parse("{\"title\": \"Unix Handbook\"}") // Parses
JSON.parse("{title: \"Unix Handbook\"}") // Syntax error
```

Not sure, but might be worth keeping in mind.